build/make/k8s-component.mk

```diff
@@ -0,0 +1,116 @@
+helm-reinstall: helm-delete helm-apply ## Uninstalls the current helm chart and reinstalls it.
+
+.PHONY: helm-chart-import
+helm-chart-import: check-all-vars check-k8s-artifact-id helm-generate-chart helm-package-release image-import ## Imports the currently available chart into the cluster-local registry.
+	@echo "Import ${K8S_HELM_RELEASE_TGZ} into K8s cluster ${K3CES_REGISTRY_URL_PREFIX}..."
+	@${BINARY_HELM} push ${K8S_HELM_RELEASE_TGZ} oci://${K3CES_REGISTRY_URL_PREFIX}/k8s ${BINARY_HELM_ADDITIONAL_PUSH_ARGS}
+	@echo "Done."
+
+##@ K8s - Helm release targets
+
+.PHONY: helm-generate-release
+helm-generate-release: ${K8S_HELM_TARGET}/templates/$(ARTIFACT_ID)_$(VERSION).yaml ## Generates the final helm chart with release urls.
+
+${K8S_HELM_TARGET}/templates/$(ARTIFACT_ID)_$(VERSION).yaml: $(K8S_PRE_GENERATE_TARGETS) ${K8S_HELM_TARGET}/Chart.yaml
+	@sed -i "s/'{{ .Namespace }}'/'{{ .Release.Namespace }}'/" ${K8S_HELM_TARGET}/templates/$(ARTIFACT_ID)_$(VERSION).yaml
+
+.PHONY: helm-package-release
+helm-package-release: ${BINARY_HELM} helm-delete-existing-tgz ${K8S_HELM_RELEASE_TGZ} ## Generates and packages the helm chart with release urls.
+
+.PHONY: helm-delete-existing-tgz
+helm-delete-existing-tgz: ## Remove an existing Helm package.
+# remove
+	@rm -f ${K8S_HELM_RELEASE_TGZ}*
+
+${K8S_HELM_RELEASE_TGZ}: ${BINARY_HELM} ${K8S_HELM_TARGET}/templates/$(ARTIFACT_ID)_$(VERSION).yaml helm-generate-chart $(K8S_POST_GENERATE_TARGETS) ## Generates and packages the helm chart with release urls.
+	@echo "Package generated helm chart"
+	@${BINARY_HELM} package ${K8S_HELM_TARGET} -d ${K8S_HELM_TARGET} ${BINARY_HELM_ADDITIONAL_PACK_ARGS}
+
+.PHONY: helm-create-temp-dependencies
+helm-create-temp-dependencies: ${BINARY_YQ} ${K8S_HELM_TARGET}/Chart.yaml
+# we use helm dependencies internally but never use them as "official" dependency because the namespace may differ
+# instead we create empty dependencies to satisfy the helm package call and delete the whole directory from the chart.tgz later-on.
+	@echo "Create helm temp dependencies (if they exist)"
+	@for dep in `${BINARY_YQ} -e '.dependencies[].name // ""' ${K8S_HELM_TARGET}/Chart.yaml`; do \
+		mkdir -p ${K8S_HELM_TARGET}/${K8S_HELM_TARGET_DEP_DIR}/$${dep} ; \
+		sed "s|replaceme|$${dep}|g" $(BUILD_DIR)/make/k8s-helm-temp-chart.yaml > ${K8S_HELM_TARGET}/${K8S_HELM_TARGET_DEP_DIR}/$${dep}/Chart.yaml ; \
+	done
+
+${BINARY_HELM}: $(UTILITY_BIN_PATH) ## Download helm locally if necessary.
+	$(call go-get-tool,$(BINARY_HELM),helm.sh/helm/v3/cmd/helm@${BINARY_HELM_VERSION})
+
+##@ K8s - Component dev targets
+
+.PHONY: component-generate
+component-generate: ${K8S_RESOURCE_TEMP_FOLDER} ## Generate the component yaml resource.
+	@echo "Generating temporary K8s component resource: ${K8S_RESOURCE_COMPONENT}"
+	@sed "s|NAMESPACE|$(ARTIFACT_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT_CR_TEMPLATE_YAML}" | sed "s|NAME|$(ARTIFACT_ID)|g"  | sed "s|VERSION|$(VERSION)|g" > "${K8S_RESOURCE_COMPONENT}"
```

Comment:
I think `$(ARTIFACT_NAMESPACE)` is not the correct namespace here.
This should be the namespace in the helm-repo. In line 62 it is hardcoded to `k8s`.